src/main/kotlin/org/rust/ide/inspections/UnnecessaryParenthesisInspection.kt

```diff
@@ -0,0 +1,35 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFixBase
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustIfExpr
+import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustParenExprImpl
+
+/**
+ * Created by zjh on 16/3/8.
+ */
+class UnnecessaryParenthesisInspection : RustLocalInspectionTool() {
+    override fun getDisplayName() = "Unnecessary parenthesis"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        object : RustVisitor() {
+            override fun visitIfExpr(expr: RustIfExpr) {
+                val exprText = expr.expr?.text
+                if (exprText?.startsWith("(")!! && exprText?.endsWith(")")!!) {
```

Comment:
i just tried `expr.expr is RustParenExpr` and the first time when i removed parentheis, the expr is still RustParenExpr only if you keep typing something, i can upload a gif to show what is going on.